### DISCUSSION TITLE
How do you handle sign up with existing verified emails?

### DISCUSSION THREAD

**Discussion by razonyang** [supabase/supabase#30511] [Questions] [answered]
Hello everyone, the SDK always returns a success response (no error) when signing up with an existing verified email.

Should this behavior be considered a bug?

If not, is there an SDK API to verify if the email was taken (or fetch user by email)? So I can handle it on my end.

## EDITED

I tried to retrieve users by email from the database directly.

```js
supabase.schema('auth').from('users').select()
```

However I got an error: `The schema must be one of the following: public, graphql_public`.

And then I found out that this approach is impossible, as the image shown below, the `auth` schema is not allowed to be exposed.

![image](https://github.com/user-attachments/assets/174db7c1-69ec-4a22-8451-0ad426922651)

**Comment by GaryAustin1** [comment, marked as answer]
It is not a bug and the way Supabase implements it for security.
You cannot access the auth schema from the API.
You can write a security definer Postgres function that you call with rpc and it checks auth.users for an email.  But doing so now creates the security leak (probe your user base by email) that Supabase was avoiding.

**Reply by razonyang** [reply]
Thank you for your explanations.
